crates/ursa-rpc-server/src/server.rs

```diff
@@ -32,19 +32,24 @@ where
         }
     }
 
-    pub async fn start(&self, config: ServerConfig) -> Result<()> {
-        info!("Server (Rpc and http) starting up");
-        let rpc_router = Router::new()
+    pub async fn start(&self, config: &ServerConfig, metrics: Option<Router>) -> Result<()> {
+        info!(
+            "Server (rpc, http{}) starting up",
+            if metrics.is_some() { " + metrics" } else { "" }
+        );
+
+        let rpc = Router::new()
             .merge(routes::network::init())
             .layer(Extension(self.rpc_server.clone()));
 
         let http = Router::new()
             .merge(http::routes::network::init::<S>())
+            .merge(metrics.unwrap_or_else(Router::new))
```

Comment:
```suggestion
            .merge(ursa_metrics::routes::init())
```

another PR, also food for through.

Reply:
Goal was to have it easy to make optional in the future if wanted